### ISSUE TITLE
[BUG] The following positioning properties are ignored: minLeft, minTop, maxLeft, maxTop 

### ISSUE THREAD

**Issue by temuri416** [Flyer53/jsPanel4#27]
Hi Stefan,

https://codepen.io/tmrczk/pen/PRzwba

minLeft, minTop, maxLeft, maxTop properties seem to be ignored. I'm able to drag the window outside of 50,50 - 350,350 area.

Thanks!

**Comment by Flyer53** [comment]
Hi Temuri,
It's not really a bug. It depends somehow on the used values for option.position.
Inside the positioning function values for minLeft, minTop etc. are considered before the offsets. In other words offsets overrule minLeft etc.
You could argue of course to look at the offsets first and then consider minLeft etc. But on the other hand it doesn't make a lot of sense to specify an offsetX value e.g. larger than a maxLeft setting. Do you have a setup where this might be useful?
But I agree that it's a point worth to think about ...

**Comment by temuri416** [comment]
ok, let's look at one thing at a time.

you're saying that "it depends somehow on the used values for option.position.". However, config like this isn't working either:

```js
jsPanel.create({
  position: {
    minLeft: 150,
    minTop: 150,
    maxLeft: 350,
    maxTop: 350
  }
});
```

so, the question is - when DO these properties work? :)

**Comment by Flyer53** [comment]
Your code sample above can't work because the positioning function doesn't know where to start to position the panel initially. The parameters `my` and `at` are missing. There is no functionality inside that fills missing `my`/`at` settings with `left: 0, top: 0` or something similar.

The positioning function starts with positioning the panel according to the parameters `my`/`at`, then the `minLeft` etc. settings are applied followed by any correction by parameter `modify` and at last the `offsets` are applied.
In short the sequence is `my`/`at` -> `minLeft` etc. -> `modify` -> `offsets`
It might be more logical to put minLeft etc. at the end. As I said ... it's worth to think about.

**Comment by Flyer53** [comment]
@temuri416 
So I made some changes to the positioning engine.
+ the order in which the positioning parameters are processed is changed
+ offsets and minLeft etc. all accept now any valid css length value
+ if used parameter modify might need a change in your code

Take a look at http://alpha.jspanel.de/docs/#options/position for details
Below all the parameter descriptions is a short block explaining the basic positioning process.
http://alpha.jspanel.de/downloads/jspanel-4-beta.zip
I personally think it's a reasonable setup now ...

**Comment by temuri416** [comment]
Thanks,

Could you also update the doc page with an example of correct use of `min*` and `max*` properties?

**Comment by temuri416** [comment]
Another question... I vaguely recall that in version 3 there was a parameter that wouldn't allow you to drag window outside min/man X or Y.

Is there something similar in v4?

Cheers!

**Comment by Flyer53** [comment]
My pleasure ...
I'm currently working on completely new documentation pages which I hope to have ready with the beta.4 release in about 3 weeks (starting tomorrow I'm out of town until April 2.). And I'm confident that I'm closing in on a final v4.

Take a look at parameter `containment` of [option dragit](http://alpha.jspanel.de/docs/#options/dragit). That's probably what you're after. The same parameter is available for [option.resizeit](http://alpha.jspanel.de/docs/#options/resizeit)

But the easiest way is the following setup:
```javascript
jsPanel.create({
    maximizedMargin: 50, // or array like [50,20,100,20] for individual top, right, bottom, left values
    syncMargins: true
});
```
See [option.syncMargins](http://alpha.jspanel.de/docs/#options/syncMargins) for a description.

**Comment by Flyer53** [comment]
Ah sorry ... you're talking about the positioning options. I had dragging and resizing in mind.

Ok, I will add another example trying to explain this a bit more clearly ...

**Comment by Flyer53** [comment]
@temuri416 
Is there anything more concerning this topic?
Otherwise be so kind and close the issue please.
Thanks
